### ISSUE TITLE
LinearAlgebra.LAPACK.geqp3!  issue

### ISSUE THREAD

**Issue by andreasvarga** [JuliaLang/julia#34780]
There is no concordance between the documentation of `geqp3!(A) `
```
"""
    geqp3!(A) -> (A, jpvt, tau)
Compute the pivoted `QR` factorization of `A`, `AP = QR` using BLAS level 3.
Returns `A`, modified in-place, `jpvt`, which represents the pivoting matrix `P`,
and `tau`, which stores the elementary reflectors.
"""
```
and the actual computed results: see the last line of the LAPACK wrapper function      
   
```
function geqp3!(A::AbstractMatrix{$elty}, jpvt::AbstractVector{BlasInt}, tau::AbstractVector{$elty})
.
.
.
       return A, tau, jpvt
end
```

**Comment by andreasnoack** [comment]
Will be fixed in https://github.com/JuliaLang/julia/pull/34784. However, I realized that the reason for this issue is that it's possible to pass preallocated `jpvt` and `tau` but there the ordering is `geqp3!(A, jpvt, tau)`. The flipped ordering of `jpvt` and `tau` for respectively input and output is pretty frustrating. We should probably try to fix that in Julia 2.0.

**Comment by andreasvarga** [comment]
Just a matter of taste: I think the ordering `A, tau, jpvt ` provides a natural  separation between the performed left and right transformations.
